src/payment/payment-action-creator.ts

```diff
@@ -12,6 +12,19 @@ import { InitializeOffsitePaymentAction, PaymentActionType, SubmitPaymentAction
 import PaymentRequestSender from './payment-request-sender';
 import PaymentRequestTransformer from './payment-request-transformer';
 
+interface InitializeOffsitePaymentSettings {
+    methodId: string;
+    gatewayId?: string;
+    instrumentId?: string;
+    target?: string;
+    promise?: Promise<undefined>;
```

Comment:
I know this was inherited, but I wonder if there's any better naming for this? What's the purpose of this promise?

Reply:
As far as I can tell it's an injectable rival promise that can be used to beat the promise of `_paymentRequestSender.initializeOffsitePayment` in a race. I think it's gonna be hard to come up with a good name as I'm not sure its the best solution for what it's trying to achieve :/

Reply:
externalPaymentInitializationPromise? 🤷